src/RequestParser.php

```diff
@@ -28,27 +37,23 @@ public function feed($data)
 
             // Extract the header from the buffer
             // in case the content isn't complete
-            list($headers, $this->buffer) = explode("\r\n\r\n", $this->buffer, 2);
+            list($headers, $buffer) = explode("\r\n\r\n", $this->buffer, 2);
```

Comment:
This might result in a large string operation. Better use the previous `strpos` and check that against the `$this->maxSize` before. You might also want to use `substr` as you already have the position then.